client/lib/query-manager/index.js

```diff
@@ -104,6 +104,27 @@ export default class QueryManager {
 	}
 
 	/**
+	 * A sorting function that defines the sort order of items under
+	 * consideration of the specified query.
+	 *
+	 * @param  {Array}  keys  Keys to be sorted
+	 * @param  {Array}  items Items by which to sort
+	 * @param  {Object} query Query object
+	 * @return {Array}        Sorted keys
+	 */
+	sort( keys, items, query ) {
+		return keys.sort( ( keyA, keyB ) => {
+			if ( ! items[ keyA ] || ! items[ keyB ] ) {
+				// One of the items has yet to be removed from the
+				// set at this point in iteration, so don't bother
+				// trying to sort.
+				return 0;
```

Comment:
>Not necessarily, since it should work fine without.

Fair enough. Should we add a clarifying comment since it might be unclear when we or another revisits this in the future?